src/column/MatrixColumn.ts

```diff
@@ -256,40 +256,52 @@ export default class MatrixColumn extends AColumn<number, INumericalMatrix> {
   }
 
 
-  private stratifyMe = (event: any, colid) => {
-    const s = this.colStratManager.updateStratifiedRanges(colid);
-    this.makeMatrixView(s);
+  private stratifyCheck() {
+    const cols = this.colStratManager.columns;
+    const categoricalCol = cols.filter((c) => c.data.desc.value.type === VALUE_TYPE_CATEGORICAL);
+    const checkColumnTie = findColumnTie(cols); // Find the index of numerical column or String
+    // If there is zero number of categorical column or the string or number as first column then the stratification is null
+    if (categoricalCol.length === 0 || checkColumnTie === 0) {
+      this.stratifyColID = null;
+      return this.makeMatrixView([this.colStratManager.nonStratifiedRange]);
+    }
+    // If there is categorical column above the numerical or string and the stratification is null then set first categorical column as stratification
+    if (categoricalCol.length > 0 && this.stratifyColID === null) {
+      this.stratifyColID = categoricalCol[0].data.desc.id;
+    }
+    const s = this.colStratManager.updateStratifiedRanges(this.stratifyColID);
+    return this.makeMatrixView(s);
   }
```

Comment:
I think the whole stratification check is already implemented. So there is no need to calculate it again.